### PR TITLE
Upgrade etcd client to 3.5

### PR DESCRIPTION
**Pull request by brandondr96**
This upgrades the etcd client to 3.5 and fixes the related dependencies which changed. Additionally, since `clientv3` was changed to `client/v3`, I switched the code to reference `v3` for clarity.

### CODE REVIEW

**Review by jswarren4 — CHANGES_REQUESTED** [review]
Requesting changes for now since we're not ready to switch to 3.5 yet.

**Comment by kramvan1** [comment]
Need to revert these 3.5 backports now https://github.com/IBM-Cloud/go-etcd-rules/pull/122

**Review by kramvan1 — CHANGES_REQUESTED** [review]
Need to revert https://github.com/IBM-Cloud/go-etcd-rules/pull/122

**Review by jswarren4 — DISMISSED** [review]
Approving so my change request doesn't block merge.

File: Makefile
Comment (kramvan1) [inline Makefile:33]: Can make this 3.5.2 now

File: rules/engine_test.go
Comment (kramvan1) [inline rules/engine_test.go:39]: Might be nice to push this into a helper function.

**Review by kramvan1 — DISMISSED** [review]

File: go.mod
Comment (sonaljain0) [inline go.mod:28]: Shouldn't this be 3.5.6?

File: go.sum
Comment (sonaljain0) [inline go.sum:160]: Lots of 3.5.7 here

**Review by sonaljain0 — CHANGES_REQUESTED** [review]
Added comments

**Review by sonaljain0 — APPROVED** [review]

**Pull request merged**
